plugins/scaffolder-backend-module-github/src/actions/githubPullRequest.ts

```diff
@@ -253,10 +254,16 @@ export const createPublishGithubPullRequestAction = (
             description:
               'Forces the author to be empty. This is useful when using a Github App, it permit the commit to be verified on Github',
           },
+          createWhenEmpty: {
+            type: 'boolean',
+            title: 'Create When Empty',
+            description:
+              'Set whether to create pull request when there are no changes to commit. The default value is true',
+          },
         },
       },
       output: {
-        required: ['remoteUrl'],
+        required: [],
```

Comment:
Was this unintentional? Looks unrelated at least.

Reply:
This is intentional. If createWhenEmpty is set to false and there are no changes to be PR'd, no PR will be created and therefore no remoteUrl will be output. Making the createWhenEmpty option available necessitates that the remoteUrl output be an optional output.

Reply:
I think this change in its form is technically breaking as users would expect this to always return. We could however split this into a `oneOf` schema instead, that when you provide the new option that the return URL is optional.

Or we just simplify that and build then URL at runtime regardless of calling the repo create call. 

Could argue that the first option even though more complex is better than trying to guess what the output might be, and might even over time be more complex.

Reply:
It is technically breaking in terms of the schema. However, it would not break any existing templates as the new `createWhenEmpty` option defaults to `true` which will throw an error if we don't have a pull request.

I favor the `oneOf` schema if that doesn't add too much overhead for the consumer to understand.

> Or we just simplify that and build then URL at runtime regardless of calling the repo create call.

Are you thinking we could set `remoteUrl` to `https://${host}/${owner}/${repo}/pulls` if no pull request is opened? We currently send Backstage notifications to the owning group when a pull request is opened. We would like to do this conditionally based on a pull request being opened. This option wouldn't allow for that.

Reply:
My understanding is that `remoteUrl` is actually not the PR link, that's what the `prNumber` is output for.

https://github.com/backstage/backstage/blob/8fb1f2ae9812084c40495d2941aa5cbf4444210b/plugins/scaffolder-backend-module-github/src/actions/githubPullRequest.ts#L454

I thought that `remoteUrl` should be for example `https://github.com/backstage/backstage`, but happy to be proven wrong on that, and if it is, then it should definitely be renamed anyways in a separate todo / breaking change.

If it is just the `repoUrl` then that's pretty easy to build always, even if a little bit icky for the most part.

Reply:
`remoteUrl` is described as "Link to the pull request in Github": https://github.com/backstage/backstage/blob/8fb1f2ae9812084c40495d2941aa5cbf4444210b/plugins/scaffolder-backend-module-github/src/actions/githubPullRequest.ts#L273-L277

I've confirmed it is indeed a link to the pull request.

Reply:
I think we should update the json schema to be `oneOf` then in this case, that if you provide `createWhenEmpty` as false, then the `remoteUrl` is no longer required.

Reply:
I've updated the output schema to utilize oneOf to allow remoteUrl to be an optional output. In the process of doing this, I notice that the Installed Actions page was not rendering oneOf output schemas. I've included a change in this PR to the scaffolder plugin  that will allow for these schemas to be displayed:
<img width="1315" alt="Screenshot 2025-02-04 at 2 47 01 PM" src="https://github.com/user-attachments/assets/80843773-1d65-4a87-b75c-9d1d270f4184" />

Reply:
I think that this is probably fine to treat as a separate issue and we fix in a followup PR :pray:

Reply:
Ended up back tracking on this a little bit, as it's still a breaking change either way as you can't switch the output based on a given input, which is an unfortunate restriction of the framework, maybe something to think about for the future.

So i just updated this PR to be breaking for now, as in practice it not breaking, only when you provide the new option.

I kept the changes for rendering though as they're still useful, but just not to this change in the PR so much anymore :pray: